### DISCUSSION TITLE
idiomatic table initialization?

### DISCUSSION THREAD

**Discussion by JLPLabs** [pallene-lang/pallene#475] [Q&A] [answered]
Context: Say I have a function that uses a couple of variables that are tables, the content of which is created during the execution of the function. How should I declare these variables and initialize them?

Is the approach below aligned with how you expect folks to address initializing tables? How is there another approach that is idiomatic?

```
local m: module = {}

typealias Point = {
    x: integer,
    y: integer,
}

function m.init(): Point
    local res: Point = {
        x = 0,
        y = 0,
    }
    return res
end


function m.add(p1: Point, p2: Point): (Point, Point)
    local res1:Point = m.init()
    local res2:Point = m.init()

    res1.x = p1.x + p2.x; res2.x = p1.x - p2.x
    res1.y = p1.y + p2.y; res2.y = p1.y - p2.y

    return res1, res2
end

return m
```

**Comment by hugomg** [comment, marked as answer]
The most idiomatic is something like you did: a table constructor that holds all the relevant fields.

Pallene doesn't like building tables incrementally. In Lua, one thing that is sometimes done is start with an empty table and then set each field using assignment statements. However, in Pallene that kind of thing isn't easy to do, because Pallene doesn't allow assignment statements to change the type of a table.

**Reply by JLPLabs** [reply]
Thanks for the quick reply. This answers my question. (I want to click on "Mark as answer" but it won't let me... is that left to you?)
